master/pkg/stream/declarative.go

```diff
@@ -146,16 +146,17 @@ func ProcessKnown(known string, exist []int64) (string, []int64, error) {
 	kok, k, err := ksi.Next()
 	xok, x := existNext()
 	for kok && xok && err == nil {
-		if k == x {
-			// ignore matched values; advance x and k
+		switch {
+		// ignore matched values; advance x and k
```

Comment:
Can we put the comments inside the case clauses?

Maybe it's just me, but I find the following commenting to be the most readable:

```
// Check for special case XYZ.
if isSpecialCase() {
    ...
}
```

OR

```
if isSpecialCase() {
    // special case detected: do something special
    ...
}
```

Comments outside of the conditional should reflect that it isn't yet known if the conditional will be true or not.  "Check for special case", for instance, suggests that we don't know yet.

Comments inside the condition should reflect that we know the conditional is true. For example, "special case detected" is only true _inside_ the conditional, not before it or after it.

The comments in this if-statement-gone-switch-statement are of the second variety "ignore matched values", "x is ahead", "k is ahead".  So I think the code is easier to read if the comments written in that style go inside the conditionals (that is after each case statement and indented one line).